### DISCUSSION TITLE
Bayesian network to formula

### DISCUSSION THREAD

**Discussion by jonasvit** [pgmpy/pgmpy#1598] [Q&A]
Hi. I wonder if there is a way to get the mathematical formula from the already built Bayesian network with pgmpy? It would be really nice when we have more complex models (not naive) and there is a need to explain final results or calculate results ourselves to check the answer.

**Comment by ankurankan** [comment]
@jonasvit I am not sure what this formula might look like. Could you maybe give an example?

**Comment by jonasvit** [comment]
Of course. Here is an example:
![image](https://user-images.githubusercontent.com/89067278/213127728-abafb4b0-4f66-4168-a6ae-bf9f2d838253.png)

We can plot our models but it would be also nice to see the formulas, how we estimate probability for a nod.

**Comment by ankurankan** [comment]
@jonasvit Since the formula is essentially the product of each conditional probability distribution (CPD), and the CPDs are defined as P(var | parents), we can write something like this to get the formula:
```python
In [7]: from pgmpy.models import BayesianNetwork

In [8]: model = BayesianNetwork([('A', 'C'), ('B', 'C')])

In [9]: ''.join([f"P({var}|{', '.join(model.get_parents(var))})" for var in model.nodes()])
Out[9]: 'P(A|)P(C|A, B)P(B|)'
```

**Reply by jonasvit** [reply]
Okay @ankurankan , but how do we get the formula of how to calculate the probability, say, even A given B and C?

**Reply by ankurankan** [reply]
@jonasvit A simple formula would basically be something like this `P(A | B, C) = P(A, B, C) / P(B, C)`. And then we would need to compute `P(A, B, C) = P(A)P(C|A)P(B)` and `P(B, C) = \sum_{A} P(A)P(C|A)P(B)`. The problem is that computing `P(B, C)` in an efficient way is an NP-Hard problem and algorithms like Variable Elimination try to compute it as efficiently as possible.

**Reply by jonasvit** [reply]
@ankurankan these formulas are exactly what I want. I just wanted to know if it is possible to print the formula for the nod that I am calculating the value, a full formula with all dependencies and nodes included

**Reply by ankurankan** [reply]
@jonasvit There isn't a function to do that in pgmpy, but you can use the code in the comment above to print it out.
